iot-hub/Quickstarts/read-d2c-messages/ReadDeviceToCloudMessages.js

```diff
@@ -1,55 +1,159 @@
+ */
+async function convertIotHubToEventHubsConnectionString(connectionString) {
+  const { HostName, SharedAccessKeyName, SharedAccessKey } = parseConnectionString(
+    connectionString
+  );
+
+  // Verify that the required info is in the connection string.
+  if (!HostName || !SharedAccessKey || !SharedAccessKeyName) {
+    throw new Error(`Invalid IotHub connection string.`);
+  }
+
+  //Extract the IotHub name from the hostname.
+  const [iotHubName] = HostName.split(".");
+
+  if (!iotHubName) {
+    throw new Error(`Unable to extract the IotHub name from the connection string.`);
+  }
+
+  // Generate a token to authenticate to the service.
+  const token = generateSasToken(
+    `${HostName}/messages/events`,
+    SharedAccessKey,
+    SharedAccessKeyName,
+    5 // token expires in 5 minutes
+  );
+  const connectionOptions = {
+    transport: "tls",
+    host: HostName,
+    hostname: HostName,
+    username: `${SharedAccessKeyName}@sas.root.${iotHubName}`,
+    port: 5671,
+    reconnect: false,
+    password: token,
+  };
+
+  const connection = new Connection(connectionOptions);
+  await connection.open();
+
+  // Create the receiver that will trigger a redirect error.
+  const receiver = await connection.createReceiver({
+    source: { address: `amqps://${HostName}/messages/events/$management` },
+  });
+
+  return new Promise((resolve, reject) => {
+    receiver.on(ReceiverEvents.receiverError, (context) => {
+      const error = context.receiver && context.receiver.error;
+      if (isAmqpError(error) && error.condition === "amqp:link:redirect") {
+        const hostname = error.info && error.info.hostname;
+        if (!hostname) {
+          reject(error);
+        } else {
+          resolve(
+            `Endpoint=sb://${hostname}/;EntityPath=${iotHubName};SharedAccessKeyName=${SharedAccessKeyName};SharedAccessKey=${SharedAccessKey}`
+          );
+        }
+      } else {
+        reject(error);
+      }
+      connection.close().catch(() => {
+        /* ignore error */
+      });
+    });
+  });
+}
 
 var printError = function (err) {
   console.log(err.message);
 };
 
 // Display the message content - telemetry and properties.
 // - Telemetry is sent in the message body
-// - The device can add arbitrary application properties to the message
+// - The device can add arbitrary properties to the message
 // - IoT Hub adds system properties, such as Device Id, to the message.
 var printMessage = function (message) {
-  console.log('Telemetry received: ');
+  console.log("Telemetry received: ");
   console.log(JSON.stringify(message.body));
-  console.log('Application properties (set by device): ')
-  console.log(JSON.stringify(message.applicationProperties));
-  console.log('System properties (set by IoT Hub): ')
-  console.log(JSON.stringify(message.annotations));
-  console.log('');
+  console.log("Properties (set by device): ");
+  console.log(JSON.stringify(message.userProperties));
```

Comment:
This field should be `properties`. Users also add properties to a message from the iot sdk using `message.properties.add(key, value)` so the fields are consistent.
```suggestion
  console.log(JSON.stringify(message.properties));
```

Reply:
Fixed with https://github.com/Azure-Samples/azure-iot-samples-node/pull/38/commits/6ec24bfeef1c20839faea9b39592f49bba2f55ac